### ISSUE TITLE
Spotify endless loop "cannot be verified"

### ISSUE THREAD

**Issue by matkoch** [Homebrew/homebrew-cask#161665]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match / SHA256 mismatch](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error (do not open an issue before trying to open a PR to fix first).

### Description of issue

I've updated Spotify about 2 weeks ago. I'm not sure I restarted my system since then, but once I did today, I'm in an endless loop of this dialog:

![2023-12-05-CleanShot-CleanShot-001443](https://github.com/Homebrew/homebrew-cask/assets/5005566/1ed4b987-897b-4cf4-8fc8-66339bbe120f)

It doesn't matter what option I choose, after 10 seconds, the dialog appears again (and steals focus).

I tried `brew uninstall` (with force), but it kept happening.

I noticed that after disabling Spotify from login items, it stops happening.

![2023-12-05-System Settings-Login Items-001445](https://github.com/Homebrew/homebrew-cask/assets/5005566/68366a87-d1f0-4e07-b3c6-7c79066d99eb)

### Command that failed

brew uninstall --cask spotify --force

### Output of command with `--verbose --debug`

```shell
...
```

### Output of `brew doctor` and `brew config`

```shell
HOMEBREW_VERSION: 4.1.23
ORIGIN: https://github.com/Homebrew/brew
HEAD: fc259153591a309e2b6fee6f971eb61b16a3b6c0
Last commit: 12 hours ago
Core tap JSON: 05 Dec 07:01 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.40.0 => /opt/homebrew/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: N/A
Xcode: 15.0.1
Rosetta 2: false
```

### Output of `brew tap`

```shell
1password/tap
adoptopenjdk/openjdk
azure/functions
homebrew/bundle
homebrew/cask-drivers
homebrew/cask-fonts
homebrew/cask-versions
iina/mpv-iina
jetbrains/utils
oven-sh/bun
powershell/tap
romkatv/powerlevel10k
sidneys/homebrew
teamookla/speedtest
yt-dlp/taps
```

**Comment by bevanjkay** [comment]
Can you please include the details from `brew config` as it helps to diagnose the issue knowing which software / hardware your system utilises.

**Comment by matkoch** [comment]
```
HOMEBREW_VERSION: 4.1.23
ORIGIN: https://github.com/Homebrew/brew
HEAD: fc259153591a309e2b6fee6f971eb61b16a3b6c0
Last commit: 12 hours ago
Core tap JSON: 05 Dec 07:01 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.40.0 => /opt/homebrew/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 14.0-arm64
CLT: N/A
Xcode: 15.0.1
Rosetta 2: false
```

**Comment by bevanjkay** [comment]
@matkoch I can replicate this issue, I'm not sure of the issue but it seems to be do with Quarantine.

**Comment by Shepard777** [comment]
Hey

**Comment by krehel** [comment]
Is this issue with the most up to date version of Spotify?  Looks like the software was updated on 24-Nov, which is after the  date of download in the original screenshot.  If not, can you update and see if this is replicated?

I wasn't able to replicate this, unfortunately. (I know, not helpful.)  I don't use Spotify anymore; I installed on a machine that has never had it.  But no issues with `brew install` and launching.

**Comment by bevanjkay** [comment]
Interestingly I have seen this issue before, but got around it with a quarantine method that I *will not* recommend.
But it seems that after uninstalling, and doing a full clean install the issue does not persist.

**Comment by krehel** [comment]
Yeah, I wasn't going to post as I didn't think it would be helpful, but thought a data point from never having it installed and being successful might point to something wrong for an upgrade workflow.

**Comment by matkoch** [comment]
@krehel the screenshots are of yesterday. I also did a `brew uninstall` and `brew install`, right before. Idk why the screenshots still mentions Nov 20.

**Comment by ringods** [comment]
I encounter this today as well with the very latest version on my Intel MBP.

<img width="372" alt="Screenshot 2023-12-06 at 19 56 16" src="https://github.com/Homebrew/homebrew-cask/assets/77923/6ba541bb-a2ea-44f0-9bfd-be96928ff08a">

But it is not the first time I have this with Spotify. I have had this with some of the previous version too. In such case, I open `System Settings` and go to `Privacy & Security`. After the previous dialog, you should see the following: 

<img width="699" alt="Screenshot 2023-12-06 at 19 59 32" src="https://github.com/Homebrew/homebrew-cask/assets/77923/3155155a-0a46-45e9-bcc0-448ed1908b45">

After clicking `Open Anyway`, you get the `macOS cannot verify...` dialog again. If you click on `Open` there, Spotify launches.

All in all, Spotify should be verified company. The steps above don't feel safe & shouldn't be needed

**Comment by matkoch** [comment]
FYI, the "Open Anyway" did NOT fix it for me. I should've included that in the OP.

**Comment by jarne** [comment]
Had the same issue here on an M1 Mac. `uninstall` and then `install` again fixed it for me (`reinstall` didn't).

**Comment by tuxpeople** [comment]
> Had the same issue here on an M1 Mac. `uninstall` and then `install` again fixed it for me (`reinstall` didn't).

Same problem here, also on an M1 Mac. `uninstall` and `install` fixed it for me as well:

```
 tdeutsch at thor in ~
$ brew uninstall spotify
==> Uninstalling Cask spotify
==> Removing launchctl service com.spotify.webhelper
Password:
==> Backing App 'Spotify.app' up to '/usr/local/Caskroom/spotify/1.2.26.1187,36b
==> Removing App '/Applications/Spotify.app'
==> Purging files for version 1.2.26.1187,36b715a1,270 of Cask spotify

 tdeutsch at thor in ~
$ brew install spotify
Running `brew update --auto-update`...
Installing from the API is now the default behaviour!
You can save space and time by running:
  brew untap homebrew/core
  brew untap homebrew/cask
==> Auto-updated Homebrew!
Updated 4 taps (aquasecurity/trivy, homebrew/cask-versions, homebrew/cask-fonts and homebrew/cask).
==> New Casks
font-playwrite-co          font-playwrite-fr-trad     sakura

You have 4 outdated formulae installed.

==> Downloading https://download.scdn.co/Spotify.dmg
###################################################################################################################################################################### 100.0%
Warning: No checksum defined for cask 'spotify', skipping verification.
==> Installing Cask spotify
==> Moving App 'Spotify.app' to '/Applications/Spotify.app'
🍺  spotify was successfully installed!
```

MacOS only asked if I really want to open this app downloaded by cask from the internet. This was a one time only dialog, next start of Spotify was without any dialogs.

**Comment by ringods** [comment]
Today, this `cannot be verified` error started to happen for cask `warp` on my M1 MacBook Pro also. An `uninstall`, `cleanup` and `install` (not `reinstall`) worked to resolve it, but it is quite annoying having to do this.

**Comment by p-linnane** [comment]
Apologies for the issue going stale, and we appreciate everyone's patience. We are looking into this and believe we have a lead on why it's happening. More to come.

**Comment by ringods** [comment]
Looking forward to the direction you are investigating, @p-linnane. Bumped into this problem again today and would really see this fixed. It is so annoying.

**Comment by bevanjkay** [comment]
Did you do anything specific before it happened again? Trying to establish how to replicate the issue consistently

**Comment by ringods** [comment]
From time to time, I upgrade all my casks using [brew-cask-upgrade](https://github.com/buo/homebrew-cask-upgrade). When I run `brew cu --all`, I see which ones need an update. A few days ago, I picked up [this new release of Spotify](https://github.com/Homebrew/homebrew-cask/commit/ccfd55506dbbf01e66c03a67d2e20b3331865af1).

Current output from `brew cu --all` for Spotify:
```
26/33  spotify                     1.2.30.1135                   1.2.30.1135                    Y   [   OK   ]
```

Since then, my Spotify setup is broken again with the `cannot be verified` error again. A `brew reinstall --cask spotify` doesn't help.

**Comment by bevanjkay** [comment]
Thanks for the info. You can fix it temporarily by running brew uninstall spotify and then brew install spotify

**Comment by ulgens** [comment]
> Thanks for the info. You can fix it temporarily by running brew uninstall spotify and then brew install spotify

This doesn't surpass the problem for me. I also had to clean the cache and restart the device to get rid of that error.

**Comment by a1rwulf** [comment]
This is a super annoying issue.
I uninstalled the app and ran `brew cleanup` but that still doesn't get rid of the message loop.
How can you get this message for an app that doesn't even exist on the system!?

**Comment by ihor-sviziev** [comment]
Same issue with spotify happened to me today. `brew uninstall --cask spotify`, `brew cleanup` and `brew install --cask spotify` helped, but it's annoying. This definitely should be fixed

**Comment by razvanazamfirei** [comment]
Thanks for your patience, everyone! We're pushing a change that hopefully fixes the issue. If people are still having an issue, please let me know. Please also provide the output of: 
```
launchctl list | grep spotify
```
as well as any Spotify-related Console logs.

**Comment by ihor-sviziev** [comment]
@razvanazamfirei, unfortunately, today got a new update, and the same issue happened.
<img width="372" alt="image" src="https://github.com/Homebrew/homebrew-cask/assets/1873745/0b436127-9d00-4f66-b147-c187aca77023">

```

➜  ~ launchctl list | grep spotify                                            ✔

26255	0	application.com.spotify.client.138935288.141593500
```

**Comment by ulgens** [comment]
I also had the issue with the latest update, but the error message appeared less often. I think it was only once, I wasn't exactly watching for it but it was definitely not frustrating as before.

**Comment by jeremie-huy** [comment]
Tried a bunch of uninstall/cleanup/update/reinstall but nothing changed.

What did it for me was to disable Spotify in the `Login Items` -> `Allow in the Background` list, the app still works so all good I guess?

**Comment by ihor-sviziev** [comment]
@razvanazamfirei, I've got a new update today, and this time, Spotify was running while I ran brew upgrade --cask.
After the upgrade, I re-launched my Spotify, and everything worked fine—it just launched. Now, I'm confused. 
This behavior may be broken when the Spotify app is not running during upgrad.

**Comment by shiftgeist** [comment]
For me a new entry below `Settings -> Gatekeeper -> Allow applications downloaded from "App Store and identified developers` appears after starting spotify with the same prompt as seen in the issue description. Where I could approve spotify. After that when starting a new option appeared to open spotify anyway.

**Comment by ihor-sviziev** [comment]
I got confused... Today got another update (spotify 1.2.36.959,04bf500c,859 -> 1.2.37.701,e66eb7bc,1226), and Spotify was stopped at that moment.
After the upgrade, I tried to open Spotify and... it just works...
Looks like it's not related to the fact that the app is running or no.

**Comment by kfkonrad** [comment]
I have the same issue, for me emptying the trash and running `brew reinstall spotify` does the trick. From what I gather there's a Spotify agent running in the background that's still the old version in the trash

I don't have a screenshot of that anymore (it's been a couple months) but I tracked down the error log in Console and that pointed to an executable in a trashed version of `Spotify.app`

**Comment by emmaLP** [comment]
Still experiencing this issue, any updates on the cause please?

<img width="272" alt="image" src="https://github.com/Homebrew/homebrew-cask/assets/10513197/367ed920-1f06-4038-a7a3-03fb2b499d56">

Brew config:
```
HOMEBREW_VERSION: 4.3.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: 43eaeca50fe3b6c755f3fc9bc6f43669e0db1039
Last commit: 5 hours ago
Core tap JSON: 24 Jun 13:03 UTC
Core cask tap JSON: 24 Jun 13:03 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.45.2 => /opt/homebrew/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.3 => /Applications/Xcode-15.3.0.app/Contents/Developer
Rosetta 2: false
```

**Comment by rarylson** [comment]
## Workaround

Just like @matkoch mentioned in the description:

> [...] after disabling Spotify from login items, it stops happening.

So just:

1) Remove Spotify

```
brew uninstall --cask spotify --force
```

2) Open System Settings > Login Items, and toggle off Spotify.

3) Click on "Cancel" in the "cannot be verified" dialog.

## Definitive fix

The root cause is that Homebrew was leaving the old Spotify login item when removing or upgrading the app. 

This seem to be fixed by [PR #170742](https://github.com/Homebrew/homebrew-cask/pull/170742), which is already merged.

**Comment by andrewbraxton** [comment]
I just hit this again, despite previously running brew uninstall spotify and seeing the startuphelper being removed. Are we sure this is fixed?

**Comment by shiftgeist** [comment]
> For me a new entry below `Settings -> Gatekeeper -> Allow applications downloaded from "App Store and identified developers` appears after starting spotify with the same prompt as seen in the issue description. Where I could approve spotify. After that when starting a new option appeared to open spotify anyway.

This still works every time for me

![image](https://github.com/user-attachments/assets/41236d2c-5c08-4029-851a-738a2aac76c9)

**Comment by Arro** [comment]
For whatever reason, the Spotify binary that hombrew is getting is not signed as from a legitimate developer.  This is confusing to me, because it seems as though people downloading it directly from their browser (and installing it the old fashioned way) aren't also having this issue.  Unless they are?  I doubt it.

**Comment by joh-klein** [comment]
Happened again today, this fixed it, but this should not be the permanent solution:

```
brew uninstall --cask spotify && brew cleanup && brew install --cask spotify
```

**Comment by noizo** [comment]
Just happen to me again.
Only `brew uninstall --cask spotify && brew cleanup && brew install --cask spotify` helps to run it again.

> ## Workaround
> 
> Just like @matkoch mentioned in the description:
> 
> > [...] after disabling Spotify from login items, it stops happening.
> 
> So just:
> 
>     1. Open System Settings > Login Items, and toggle off Spotify.
> 
>     2. Click on "Cancel" in the "cannot be verified" dialog.
> 
> 
> ## Definitive fix
> 
> The root cause is that Homebrew was leaving the old Spotify login item when removing or upgrading the app.
> 
> This seem to be fixed by [PR #170742](https://github.com/Homebrew/homebrew-cask/pull/170742), which is already merged.
> 
> After this fix, when removing Spotify, you should see the startup service being successfully removed:
> 
> ```
> $ brew uninstall --cask spotify
> [...]
> ==> Removing launchctl service com.spotify.client.startuphelper
> [...]
> ```
> 
> Now you can just reinstall Spotify.

Mentioned definite fix is not definite.

**Comment by lantrix** [comment]
> Just happen to me again. Only `brew uninstall --cask spotify && brew cleanup && brew install --cask spotify` helps to run it again.
> Mentioned definitive fix is not definite.

Same, I'm still getting it and this is the only workaround.

**Comment by artmg** [comment]
I also tried `brew uninstall --cask spotify && brew cleanup && brew install --cask spotify` to no avail. When I do

* `brew uninstall --cask spotify && brew cleanup`
* restart computer
* `launchctl list | grep spotify`
   -  `1878	-9	com.spotify.client.startuphelper`

so the helper is still there!

My solution was to do BOTH workarounds:

* `brew uninstall --cask spotify && brew cleanup`
* Restart computer
* Click on "Cancel" in the "cannot be verified" dialog.
* Open System Settings > Login Items, and toggle off Spotify.
* `brew install --cask spotify`

That may be belt and braces but it did seem to work

**Comment by noizo** [comment]
3 weeks later.
![Screenshot 2024-09-07 at 13 49 04](https://github.com/user-attachments/assets/cfb2ddc7-296f-4f5d-8c17-fc56cf094838)

**Comment by bevanjkay** [comment]
@noizo Can you remember if Spotify was open when you ran the command?

**Comment by noizo** [comment]
@bevanjkay 
So on 5th of September Spotify was running and playing.
I run `brew upgrade --greedy && brew cleanup` and Spotify has updated. Since that day i had no Mac restarts, only sleep mode until yesterday i shut it down.
Spotify is in autostart, today i launched my Mac, result is on screenshot.
More over, this window prevents me from manual start of any other application.
Canceling is in constant loop, until i click Move to bin.

**Comment by bevanjkay** [comment]
Thanks for the info @noizo
It seems that a computer restart might be key in replicating the issue - but this isn't consistent for me.
The biggest hurdle for me has been not being able to replicate it regularly which has made this difficult to chase down.

I'm trying another `launchctl` removal, but I'm not convinced it will fix the issue - we will know after the next rounds of updates.

**Comment by Arro** [comment]
Not sure if you've tried this, but it happens when I do a `brew upgrade` . so you could uninstall, install the previous version, then `brew upgrade spotify`.

**Comment by bevanjkay** [comment]
@arro Definitely seems like the best way to go, I just haven't had access to a previous version of the Spotify application. I have a backup of the current one now, so can do some testing when there's another version bump.

**Comment by ihor-sviziev** [comment]
Just want to update - I got the same issue today, after upgrading to macOS Sequoia.  Before that, I hadn'y any issues for a few months.
Not sure if that's a new issue, but looks the same as previously.

**Comment by ulgens** [comment]
>  I got the same issue today, after upgrading to macOS Sequoia. Before that, I hadn't any issues for a few months.

It's exactly the same case here. Uninstall, clean purge and install solved it this time, but it was still annoying.

**Comment by bevanjkay** [comment]
Interesting finding @noizo - this would suggest that the launchctl item is not being successfully removed from the cask.
`launchctl remove com.spotify.client.startuphelper` - this command is ran during uninstall.

**Comment by ihor-sviziev** [comment]
Today, I've got the same issue just after the MacOS update to 15.1, no Spotify updates were recently. Maybe that could help somehow to identify the issue

**Comment by noizo** [comment]
After upgrading to 15.1.1 and shutting down mac.
![Screenshot 2024-11-25 at 08 49 56](https://github.com/user-attachments/assets/5b2030c2-7fc5-43ee-bedf-9a58a80c1293)

**Comment by lantrix** [comment]
I encounter this on a Mac where my employer also tries to update apps out of band or brew using another tool.
The same fix above works, but it may be possible it's where the app its updated by the other tool that it then becomes corrupted.

**Comment by hulmaker** [comment]
Still experiencing this issue. [noizo's comment](https://github.com/Homebrew/homebrew-cask/issues/161665#issuecomment-2363665157) worked. But it's just super inconvenient having to put in a special care for a single package.

**Comment by lantrix** [comment]
> But it's just super inconvenient having to put in a special care for a single package.

For over 1.5 years too... I just removed it from brew and installed directly.

**Comment by bevanjkay** [comment]
It looks like the application is not correctly signed, likely to do with the embedded Chromium framework. Testing the application directly from the binary fails.
```
syspolicy_check distribution /Volumes/Spotify/Spotify.app

---------------------------------------------------------------
Notary Ticket Missing
    File: Spotify.app
    Severity: Fatal
    Full Error: A Notarization ticket is not stapled to this application.
    Type: Distribution Error
    Suggested Fix: If this application has already been uploaded to the Apple
        notary service, please make sure to attach the ticket with the `stapler
        staple` command. If not, please upload to the Apple notary service
        using Xcode or via `notarytool`.
    Documentation:
        https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution.
    NSError.userInfo: ["SyspolicyCheckAdditionalInformation": ,
        "SyspolicyCheckShortError": Notary Ticket Missing,
        "SyspolicyCheckErrorFile": /Applications/Spotify.app,
        "SyspolicyCheckAdvice": If this application has already been uploaded
        to the Apple notary service, please make sure to attach the ticket with
        the `stapler staple` command. If not, please upload to the Apple notary
        service using Xcode or via `notarytool`. , "SyspolicyCheckLongError": A
        Notarization ticket is not stapled to this application.,
        "SyspolicyCheckDocumentationLink":
        https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution.,
        "SyspolicyCheckErrorLevel": Fatal]

---------------------------------------------------------------

```

The installer that can be downloaded upstream likely bypasses this.

<img width="938" height="347" alt="Image" src="https://github.com/user-attachments/assets/c8c1b6db-57b4-4e41-83d6-ee7e9ead1bb7" />

**Comment by ssbarnea** [comment]
Maybe that is all we need to move away from that service. Clearly is something new.

**Comment by p-linnane** [comment]
Closing this since it's been almost 2 years and we cannot fix upstream packaging.
